traffic_ops/traffic_ops_golang/auth/authorize.go

```diff
@@ -63,26 +64,27 @@ type key int
 
 const CurrentUserKey key = iota
 
-// GetCurrentUserFromDB  - returns the id and privilege level of the given user along with the username, or -1 as the id, - as the userName and PrivLevelInvalid if the user doesn't exist.
-func GetCurrentUserFromDB(DB *sqlx.DB, CurrentUserStmt, user string, timeout time.Duration) CurrentUser {
+// GetCurrentUserFromDB  - returns the id and privilege level of the given user along with the username, or -1 as the id, - as the userName and PrivLevelInvalid if the user doesn't exist, along with a user facing error, a system error to log, and an error code to return
+func GetCurrentUserFromDB(DB *sqlx.DB, CurrentUserStmt, user string, timeout time.Duration) (CurrentUser, error, error, int) {
 	var currentUserInfo CurrentUser
 	if DB == nil {
-		log.Errorf("no db provided to GetCurrentUserFromDB")
-		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}
+		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}, errors.New("internal server error"), errors.New("no db provided to GetCurrentUserFromDB"), http.StatusInternalServerError
```

Comment:
IMO programmers should always treat the value as undefined if an error exists, so `CurrentUser{}` should be fine here. But, not a big deal to ensure it's invalid either.

Reply:
right, that was a holdover from before this function returned errors, I will refactor that.